src/tezos_rpc/http.ml

```diff
@@ -0,0 +1,41 @@
+open Helpers
+open Error
+
+type 'a method_ =
+  | GET : unit method_
+  | POST : string method_
+let http_request (type a) ~uri ~(method_ : a method_) (data : a) =
+  let open Piaf in
+  let%await response =
+    match method_ with
+    | GET -> Client.Oneshot.get uri
+    | POST ->
+      let body = Body.of_string data in
+      Client.Oneshot.post ~body uri in
+  match response with
+  | Ok response -> (
+    let%await body = Piaf.Body.to_string response.body in
+    match body with
+    | Ok body -> await (Ok body)
+    | Error err -> await (Error (Piaf_body err)))
+  | Error err -> await (Error (Piaf_request err))
+
+let http_request ~node_uri ~path ~method_ response_of_yojson data =
+  let uri = Uri.with_path node_uri path in
+  let%await body = http_request ~uri ~method_ data in
```

Comment:
```suggestion
  let%await body = http_request uri ~method_ data in
```

Reply:
If I understand correctly, of course 😅

Reply:
This typed, as you can see from the CI. But also nope, this implementation is correct.

https://github.com/marigold-dev/deku/pull/444/files#diff-698e4c768874ed323e94fa37737b6039c3ef8f7a6f5222ecd43cfaab42d59c3bR7

Reply:
Thanks, I guess I had something weird with my environment, since it showed me `unused variable`. Working great now. Thanks!